ratis-examples/src/main/java/org/apache/ratis/examples/filestore/FileStore.java

```diff
@@ -171,6 +182,17 @@ Path resolve(Path relative) throws IOException {
     return full;
   }
 
+  CompletableFuture<ReadReplyProto> watch(String relative) {
+    final FileInfo info = files.watch(relative);
+    final ReadReplyProto reply = ReadReplyProto.newBuilder()
+        .setResolvedPath(FileStoreCommon.toByteString(info.getRelativePath()))
+        .build();
+    if (info instanceof FileInfo.Watch) {
+      return ((FileInfo.Watch) info).getFuture().thenApply(uc -> reply);
+    }
```

Comment:
1. `info instanceof FileInfo.Watch` may be an assertion.
2. We can build the watch reply based on the future payload `uc` instead of pre-build one.

Reply:
1. It can't be an assertion since `info` can also be `UnderConstruction` or `ReadOnly`.  We should return a completed future in that case.
2. Since we don't need anything from `uc`, we may use the same `reply` for both cases.